pkg/apis/imageregistry/v1alpha1/types.go

```diff
@@ -42,14 +42,17 @@ type ImageRegistryConfigStorageS3 struct {
 	Region         string `json:"region,omitempty"`
 	RegionEndpoint string `json:"regionEndpoint,omitempty"`
 	Encrypt        bool   `json:"encrypt,omitempty"`
+	ManagedStorage bool   `json:"managedStorage,omitempty"`
```

Comment:
this needs to be part of status, not part of spec.  spec is things the user controls.

Reply:
I was expecting a single field, like:  status.ManagedStorage   or possibly status.Storage.ManagedStorage

we don't need a unique field for each storagetype struct.

Reply:
I started doing that initially, but the Storage.ManagedStorage field would not be easily accessible during the CompleteConfiguration phase of configuration the storage as d.Config only holds the specific information for that storage driver.  This seemed like a cleaner way to do it.

Reply:
it has to be part of status regardless.  I can buy an argument for putting the field on each storage-type specific struct, but it has to be a status struct.

Reply:
updated per your suggestion